### PR TITLE
Agent Playing against each other

### LINKED ISSUE #63: make agent play against each other

### PR DESCRIPTION
**Pull request by pwyq**
close #63

### CODE REVIEW

**Comment by pwyq** [comment]
```
...*...*...
Turn 1: B (PNS)  plays 2 0 1
...*...*1..
Turn 2: W (Nega) plays 2 1 2
...*...*12.
Turn 3: B (PNS)  plays 2 2 2
...*...*122
Turn 4: W (Nega) plays 1 0 2
...*2..*122
Turn 5: B (PNS)  plays 0 2 2
..2*2..*122
Turn 6: W (Nega) plays 0 1 2
.22*2..*122
Turn 7: B (PNS)  plays 1 1 1
.22*21.*122
Turn 8: W (Nega) resigns.
```

**Comment by pwyq** [comment]
```
...*...*...
Turn 1: W (Nega) plays 1 1 2
...*.2.*...
Turn 2: B (PNS)  plays 0 2 2
..2*.2.*...
Turn 3: W (Nega) plays 2 2 2
..2*.2.*..2
Turn 4: B (PNS)  plays 2 0 2
..2*.2.*2.2
Turn 5: W (Nega) plays 2 1 1
..2*.2.*212
Turn 6: B (PNS)  plays 0 0 1
1.2*.2.*212
Turn 7: W (Nega) resigns.
```

File: test/other_tests/twoD_vs_oneD.cpp [hunk 1]
```diff
@@ -126,6 +126,8 @@ void OneD::set(uint8_t row, uint8_t col, uint8_t val) {
 
 ////////////////////
 
+/* TODO: comment out because can't have two main(); think of a way to let all tests works.
```
Comment (pwyq) [inline test/other_tests/twoD_vs_oneD.cpp:129]: @Samarium150 this

Comment (pwyq) [reply test/other_tests/twoD_vs_oneD.cpp:129]: should we care, or just manually change main() entrance every time

Comment (Samarium150) [reply test/other_tests/twoD_vs_oneD.cpp:129]: > should we care, or just manually change main() entrance every time

we can create multiple executables

**Pull request merged**